bin/commands/internal/start/component/index.ts

```diff
@@ -0,0 +1,110 @@
+/*
+  This program and the accompanying materials are made available
+  under the terms of the Eclipse Public License v2.0 which
+  accompanies this distribution, and is available at
+  https://www.eclipse.org/legal/epl-v20.html
+ 
+  SPDX-License-Identifier: EPL-2.0
+ 
+  Copyright Contributors to the Zowe Project.
+*/
+
+import * as std from 'std';
+import * as os from 'os';
+import * as zos from 'zos';
+import * as common from '../../../../libs/common';
+import * as config from '../../../../libs/config';
+import * as shell from '../../../../libs/shell';
+import * as varlib from '../../../../libs/var';
+import * as java from '../../../../libs/java';
+import * as fs from '../../../../libs/fs';
+import * as component from '../../../../libs/component';
+import { PathAPI as pathoid } from '../../../../libs/pathoid';
+
+export function execute(componentId: string, runInBackground: boolean=false) {
+  std.setenv('ZWE_CLI_PARAMETER_COMPONENT', componentId);
+  
+  const coreComponents = std.getenv('ZWE_PRIVATE_CORE_COMPONENTS_REQUIRE_JAVA');
+  if (coreComponents && coreComponents.includes(componentId)) {
+    java.requireJava();
+  }
+
+  //node.requireNode(); //needed for what?
+  common.requireZoweYaml();
+
+  const ZOWE_CONFIG=config.getZoweConfig();
+
+  const runtimeDirectory = ZOWE_CONFIG.zowe.runtimeDirectory;
+
+  // overwrite ZWE_PRIVATE_LOG_LEVEL_ZWELS with zowe.launchScript.logLevel config in YAML
+  if (ZOWE_CONFIG.zowe.launchScript) {
+    std.setenv('ZWE_PRIVATE_LOG_LEVEL_ZWELS', ZOWE_CONFIG.zowe.launchScript.logLevel.toUpperCase());
+  };
+
+  // check and sanitize ZWE_CLI_PARAMETER_HA_INSTANCE
+  config.sanitizeHaInstanceId();
+
+  const WORKSPACE_DIRECTORY = ZOWE_CONFIG.zowe.workspaceDirectory;
+  if (!WORKSPACE_DIRECTORY) {
+    common.printErrorAndExit("Error ZWEL0157E: Zowe workspace directory (zowe.workspaceDirectory) is not defined in Zowe YAML configuration file.", undefined, 157);
+  }
+
+  // TODO really this should use configmgr
+  // load environment
+  config.loadEnvironmentVariables(componentId);
+
+  // find component root directory and execute start script
+  const COMPONENT_DIR = component.findComponentDirectory(componentId);
+  common.printFormattedTrace("ZWELS", "zwe-internal-start-component", `- found ${componentId} in directory ${COMPONENT_DIR}`);
+  if (COMPONENT_DIR) {
+    let dir = COMPONENT_DIR;
+    const manifest = component.getManifest(COMPONENT_DIR);
+    
+    const privateWorkspaceDir=std.getenv('ZWE_PRIVATE_WORKSPACE_ENV_DIR');
+    //TODO CLI_PARAMETER vars come from cli --parameters, and so should be mapped to execute() parameters. This probably doesnt work.
+    const haInstance = std.getenv('ZWE_CLI_PARAMETER_HA_INSTANCE');
+    
+    // source environment snapshot created by configure step
+    const componentName = pathoid.basename(COMPONENT_DIR);
+
+    if (fs.fileExists(`${privateWorkspaceDir}/${componentName}/.${haInstance}.env`)) {
```

Comment:
should put this within the start script check because, if there is no start script, is there really a reason to load the env vars?

Reply:
are the vars loaded for this component visible to another component? with configmgr? without?

Reply:
the difference of with/without configmgr does not matter here, checked the behavior by exporting vars from one component to another and seeing that with launcher, env vars are not shared, but without launcher they are.